.github/workflows/docker.yml

```diff
@@ -0,0 +1,88 @@
+name: Build and publish docker images
+on:
+  workflow_call:
+    inputs:
+      platforms:
+        description: > 
+          Specify which platforms to build an image for. Forwarded to 
+          docker/build-push-action@v2.
+        type: string
+        required: false
+        default: linux/amd64,linux/arm64
+      image-name:
+        description: >
+          What to call the docker image. Does not include a tag name. You need to 
+          include the 'matrixdotorg/' prefix.
+        type: string
+        required: true
+      dockerfile:
+        description: Path to the Dockerfile, relative to the repository root.
+        type: string
+        required: false
+        default: Dockerfile
+      publish:
+        description: If true, publish to dockerhub.
+        type: boolean
+        required: false
+        default: false
+    secrets:
+      DOCKER_HUB_USERNAME:
+        required: true
+      DOCKER_HUB_TOKEN:
+        required: true
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        if: contains(inputs.platforms, 'linux/arm64')
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: arm64
```

Comment:
Looks questionable, given that additional platforms may be specified in `inputs.platforms`.  
Unless the intent is for platforms to be a subset of { linux/amd64, linux/arm64 }?

Reply:
Ahh, I hadn't grokked that there are other platforms available. https://github.com/docker/setup-qemu-action lists e.g. RISCV, older ARMs, MIPS...

I don't think we're planning to use any of these apart from arm64 imminently, but we may as well keep the door open. I guess I could change the condition to setup qemu if and only if `inputs.platforms != 'linux/amd64'?

Reply:
The
```
        with:
          platforms: arm64
```
bit is going to be troublesome IMO. Unless we can do some magic?

Reply:
Maybe it's just easier to say we only suppose amd64 and arm64. If we want something more complicated we can revisit this later.

Reply:
On third thought, maybe it's easier to just keep it simple and always build amd64+arm64 and not provide the option to customise this. [Olivier was happy with this](https://github.com/matrix-org/backend-meta/pull/2#discussion_r804602636)... and if it's good enough for Synapse, it should be good enough for all our projects.